File: src/network/ClientService.java
package network;
import java.io.IOException;
import java.net.Socket;

import com.google.gson.Gson;

import led.LEDStripManager;

public class ClientService extends Thread implements ReceiveCallback {
	
	static int mNextId = 0;
	public int mId = 0;
	
	Server mServer;
	Socket mSocket;
	public Sender mSender;
	Receiver mReceiver;
	
	public ClientService(Server _server, Socket _socket) {
		mId = mNextId;
		mNextId++;
		
		mServer = _server;
		mSocket = _socket;
		
		try {
			mSender = new Sender(mSocket.getOutputStream());
			mReceiver = new Receiver(mSocket, mSocket.getInputStream(), this);
		} catch (IOException e) {
			e.printStackTrace();
		}
		
		mSender.send(new Gson().toJson(LEDStripManager.mLEDStatus));
		
		mReceiver.start();
	}

	@Override
	public void onReceiveMessage(String _message) {
		mServer.receiveMessage(_message);
	}
}

File: src/network/ClientService.kt
package network

import java.net.Socket
import java.lang.Thread
import network.ReceiveCallback
import network.Sender
import network.ClientService
import java.io.IOException
import com.google.gson.Gson
import led.LEDStripManager
import com.google.gson.annotations.SerializedName
import procedures.ProcedureType
import led.LEDDataBundle
import network.Receiver.ClientType
import kotlin.jvm.Synchronized
import kotlin.Throws
import java.net.ServerSocket

class ClientService(_server: Server, _socket: Socket) : Thread(), ReceiveCallback {
    var mId = 0
    var mServer: Server
    var mSocket: Socket
    var mSender: Sender? = null
    var mReceiver: Receiver? = null
    override fun onReceiveMessage(_message: String?) {
        println("Message received")
        mServer.receiveMessage(_message)
    }

    companion object {
        var mNextId = 0
    }

    init {
        mId = mNextId
        mNextId++
        mServer = _server
        mSocket = _socket
        try {
            mSender = Sender(mSocket.getOutputStream())
            mReceiver = Receiver(mSocket, mSocket.getInputStream(), this)
        } catch (e: IOException) {
            e.printStackTrace()
        }
        mSender!!.send(Gson().toJson(LEDStripManager.mLEDStatus))
        mReceiver!!.start()
    }
}